src/System.Text.Json/tests/Serialization/JsonElementTests.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public class JsonElementTests
+    {
+        [Fact]
+        public void SerializeJsonElement()
+        {
+            var obj = JsonSerializer.Parse<JsonElementClass>(JsonElementClass.s_json);
+            obj.Verify();
+            string reserialized = JsonSerializer.ToString(obj);
+
+            // Properties in the exported json will be in the order that they were reflected, doing a quick check to see that
+            // we end up with the same length (i.e. same amount of data) to start.
+            Assert.Equal(JsonElementClass.s_json.StripWhitespace().Length, reserialized.Length);
+
+            // Shoving it back through the parser should validate round tripping.
+            obj = JsonSerializer.Parse<JsonElementClass>(reserialized);
+            obj.Verify();
+        }
+
+        public class JsonElementClass : ITestClass
+        {
+            public JsonElement Number { get; set; }
+            public JsonElement True { get; set; }
+            public JsonElement False { get; set; }
+            public JsonElement String { get; set; }
+            public JsonElement Array { get; set; }
+            public JsonElement Object { get; set; }
+
+            public static readonly string s_json =
+                @"{" +
+                    @"""Number"" : 1," +
+                    @"""True"" : true," +
+                    @"""False"" : false," +
+                    @"""String"" : ""Hello""," +
+                    @"""Array"" : [2, false, true, ""Goodbye""]," +
+                    @"""Object"" : {}" +
+                @"}";
+
+            public static readonly byte[] s_data = Encoding.UTF8.GetBytes(s_json);
+
+            public void Initialize()
+            {
+                Number = JsonDocument.Parse(@"1").RootElement.Clone();
```

Comment:
Can you also add "null" and array-in-object/object-in-array cases?

Reply:
Null doesn't work yet. I'll add a comment. Nested any deeper doesn't matter as far as the implementation goes, I do have a more complicated nesting in the object tests just for extra sanity.

Reply:
Actually not entirely sure if we should support it. It fails now as we return `null`, which you can't do for `JsonElement`. Might be a bit weird to have null go into `object` fields (when all other values are `JsonElement`) and `JsonElement` with a Null value going into `JsonElement`. Topic for discussion. :)